packages/core/src/ceramic.ts

```diff
@@ -632,6 +633,12 @@ export class Ceramic implements CeramicApi {
     const walkNext = async (node: TrieNode, streamId: StreamID | CommitID) => {
       let stream
       try {
+        if (query.genesis) {
```

Comment:
So if a user tries to use this functionality with a non-deterministic genesis commit, it will be a signed commit. In that case I believe loading the stream will still time out because the payload won't be put into ipfs, just the jwt envelope will be.  We should add a test for that and reproduce the behavior, and then probably the thing to do is to check here if the commit looks like a signed commit (can use [StreamUtils.isSignedCommit](https://github.com/ceramicnetwork/js-ceramic/blob/6871b7dcd27d08a727ae492754440309a563efc3/packages/common/src/utils/stream-utils.ts#L213)) and if so throw an exception.

Reply:
I'm confused about how would we know if it is a signed commit before we do actually load it. But it will end up throwing a different error in that case as genesis CID =/= streamId CID

Reply:
Nvm, we can do this. Pushing it out soon